### PR TITLE
[fix]: [CDS-79548]: fetch templateType for stage with no templateInputs

### PR DESCRIPTION
**Pull request by Abhishek-CDC**
## Describe your changes
While migrating a pipeline with stage using stageTemplate when stageTemplate doesn't have runtimeInputs, it was throwing NPE as there were no templateInputs.
Making template API call for fetching type of template when there is no templateInput.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53009)

### CODE REVIEW

**Comment by Abhishek-CDC** [comment]
trigger uts

**Comment by Abhishek-CDC** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/java/io/harness/ng/core/migration/serviceenvmigrationv2/ServiceEnvironmentV2MigrationService.java [hunk 1]
```diff
@@ -804,25 +805,57 @@ private void checkStageTemplateExistence(
...
-          .getNode()
-          .getCurrJsonNode()
-          .textValue();
+      if (stageHasTemplateInputs(stageNode)) {
+        return stageNode.getField("template")
+            .getNode()
+            .getField("templateInputs")
+            .getNode()
+            .getField("type")
+            .getNode()
```
Comment (PragyeshMishra01) [inline 120-ng-manager/src/main/java/io/harness/ng/core/migration/serviceenvmigrationv2/ServiceEnvironmentV2MigrationService.java:851]: pls check for NPE

**Review by PragyeshMishra01 — COMMENTED** [review]
added comment

File: 120-ng-manager/src/main/java/io/harness/ng/core/migration/serviceenvmigrationv2/ServiceEnvironmentV2MigrationService.java (hunk 1 above)
Comment (Abhishek-CDC) [reply 120-ng-manager/src/main/java/io/harness/ng/core/migration/serviceenvmigrationv2/ServiceEnvironmentV2MigrationService.java:851]: @PragyeshMishra01 This is old code. 
Actually this PR is to address this NPE itself.
In the if condition above, we check if stage has template inputs. if there are template inputs then we get type.
else outside if block we are making API call to fetch template and get the type

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53009)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by PragyeshMishra01 — APPROVED** [review]

**Comment by Abhishek-CDC** [comment]
trigger review

**Pull request merged**